### PR TITLE
More documentation and flexibility for FTP imports.

### PR DESCRIPTION
**Pull request by jmchilton**
- Add missing documentation for the `ftp_upload_dir_identifier` option. f95cbf8f38801f03ba1ea9c065bc1796253d6eaf
- Add `ftp_upload_dir_template` option to configure how to combine identifier and base directory to generate an ftp upload directory. f95cbf8f38801f03ba1ea9c065bc1796253d6eaf
- Add option to prevent FTP uploads from being deleted on import (`ftp_upload_purge`). 997b53f3ae7586d4a2e90f6cb03f00b844bb4d42
- Refactoring to use centralized logic for determining a user's FTP upload directory 54173807e56e8ae4bef4cc0511249d6b87380de2.

### CODE REVIEW

**Comment by hexylena** [comment]
> Add option to prevent FTP uploads from being deleted on import (ftp_upload_purge). 997b53f

well that's everything I've wanted to see fixed and more. Thanks @jmchilton :D

File: config/galaxy.ini.sample [hunk 1]
```diff
@@ -1013,13 +1013,27 @@ use_interactive = True
 # database for authentication) and set the following two options.
 
 # This should point to a directory containing subdirectories matching users'
-# email addresses, where Galaxy will look for files.
+# identifer (defaults to e-mail), where Galaxy will look for files.
```
Comment (nsoranzo) [inline config/galaxy.ini.sample]: s/identifer/identifier/

File: config/galaxy.ini.sample [hunk 2]
```diff
@@ -1013,13 +1013,27 @@ use_interactive = True
...
 # users in the help text.
 #ftp_upload_site = None
 
+# User attribute to use as subdirectory in calculating default ftp_upload_dir 
+# pattern. By default this will be email so a user's FTP upload directory will be
+# ${ftp_upload_dir}/${user.email}. Can set this to other attributes such as id or
+# username though.
+#ftp_upload_dir_identifier = email
+
+# Python string tempalte used to determine an FTP upload directory for a
```
Comment (nsoranzo) [inline config/galaxy.ini.sample]: s/tempalte/template/

**Comment by nsoranzo** [comment]
@jmchilton LGTM, thanks for adding the documentation!

**Comment by nsoranzo** [comment]
:+1:

**Pull request merged**